.github/workflows/build-magma-windows.yml

```diff
@@ -0,0 +1,74 @@
+name: Build MAGMA for Windows
+
+on:
+  push:
+    branches:
+      main
+    paths:
+      - .github/scripts/windows/*
+      - .github/workflows/build-magma-windows.yml
+  pull_request:
+    paths:
+      - .github/scripts/windows/*
+      - .github/workflows/build-magma-windows.yml
+
```

Comment:
```suggestion
    permissions:
      id-token: write
      contents: read
```

Reply:
The `id-token` permission is only needed by the `push` job, so I added it on L48 (see below)